pkg/intoto/slsa_provenance/v0.2/provenance.go

```diff
@@ -200,3 +201,20 @@ func GetMaterialsGitURI(pred ProvenancePredicate) []string {
 	}
 	return gitURIs
 }
+
+// SetSLSAv02ProvenanceData sets data to verify a SLSA v02 provenance in the given ProvenanceIR.
+func SetSLSAv02ProvenanceData(provenanceIR *types.ProvenanceIR) error {
```

Comment:
Same as above, I suggest renaming this to `ParseProvenanceData`, and changing it to take a byte array as input.